script.js

```diff
@@ -1,51 +1,82 @@
+const matchEpisodes = document.getElementById("matching-ep")
+
+const allEpisodes =  getAllEpisodes();
 
-  const allEpisodes = getAllEpisodes();
-  showAllEpisodes(allEpisodes)
+function setup() {
+    showAllEpisodes(allEpisodes);
+    populateSelectedEpisode(allEpisodes)
 }
-window.onload = setup;
 
 // function makePageForEpisodes(episodeList) {
 //   const rootElem = document.getElementById("root");
 //   rootElem.textContent = `Got ${episodeList.length} episode(s)`;
 // }
 
+// show all episodes
 function showAllEpisodes(episodes){
   const container = document.querySelector(".container");
+  container.innerHTML = "";
+  totalEpisodes.textContent = allEpisodes.length; 
 
   for(let episode of episodes){
-    const divArticle = document.createElement("Article");
-
-    const title = document.createElement("h1");
-    title.style.border = "solid grey";
-    title.style.padding = "10px"
-    title.textContent = `${episode.name} - S0${episode.season}E0${episode.number}`;
-    
-    divArticle.appendChild(title);
+    const article = document.createElement("Article");
+  
+    const title = document.createElement("h2");
+    let formattedEpNumber = ((episode.number.toString()).padStart(2,"0"))// format number to two characters string
+    title.textContent = `${episode.name} - S0${episode.season}E${formattedEpNumber}`;
+    article.appendChild(title);
     
     const image = document.createElement("img");
     image.src = episode.image.medium;
-    image.style.width = "auto"
-    divArticle.appendChild(image);
+    article.appendChild(image);
 
     const p = document.createElement("p");
     p.innerHTML = `${episode.summary}`;
-    divArticle.appendChild(p);
+    article.appendChild(p);
 
-    container.appendChild(divArticle);
+    container.appendChild(article);
   }
 }
 
-function searchEpisode(episodes){
-  const div = document.createElement("div")
+//search episode
+search.addEventListener('keyup', e => {
+  const matchingValue = e.target.value.toUpperCase();
+  const matchingEpisodes = allEpisodes.filter((episode)=>{
+        return episode.name.toUpperCase().includes(matchingValue) || episode.summary.toUpperCase().includes(matchingValue);
+  })
 
-  for(let episode of episodes){
-    const search = document.createComment("input")
+  matchEpisodes.innerText = matchingEpisodes.length;
+  showAllEpisodes(matchingEpisodes);
+})
 
-    search.appendChild(div)
-  }
+//select options
+function populateSelectedEpisode(episodes){
+ const select = document.getElementById("episode-list");
+
+ for(let episode of episodes){
+  const option = document.createElement("option");
+  
+  option.setAttribute("value",episode.id);
+  let formattedEpNumber = ((episode.number.toString()).padStart(2,"0"))// format number to two characters string
+  option.innerText = `S0${episode.season}E${formattedEpNumber} - ${episode.name}`;
+  
+  select.appendChild(option);
+ }
 }
 
+//populate selected episode
+// select.addEventListener('change', e => {
```

Comment:
Try to avoid leaving commented code - I know it's tempting, but realistically, there's probably a reason you're not using it, and it will only make your filed harder to read!